### PR TITLE
TPC-H queries

### PR DESCRIPTION
**Pull request by hadia206**

### CODE REVIEW

File: examples/tpch/bodo_queries.py [hunk 1]
```diff
@@ -0,0 +1,964 @@
...
+def load_customer(data_folder):
+    data_path = data_folder + "/customer.pq"
+    df = pd.read_parquet(
+        data_path,
+    )
+    return df
+
+
+# Nation is a very small file so set it to replicated
+@bodo.jit(returns_maybe_distributed=False)
```
Comment (ehsantn) [inline examples/tpch/bodo_queries.py:91]:
```suggestion
@bodo.jit(distributed=False)
```

File: examples/tpch/bodo_queries.py [hunk 2]
```diff
@@ -0,0 +1,964 @@
...
+def load_nation(data_folder):
+    data_path = data_folder + "/nation.pq"
+    df = pd.read_parquet(
+        data_path,
+    )
+    return df
+
+
+# Region is a very small file so set it to replicated
+@bodo.jit(returns_maybe_distributed=False)
```
Comment (ehsantn) [inline examples/tpch/bodo_queries.py:101]:
```suggestion
@bodo.jit(distributed=False)
```

**Review by ehsantn — APPROVED** [review]
Looks good to me. Thanks for the effort. Just a couple of minor comments.

File: examples/tpch/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# TPC-H Queries 
+
+TPC-H is a benchmark for business oriented ad-hoc queries that is used to simulate real questions and is usually used to benchmark performance of database tools for answering them.
```
Comment (ritwika314) [inline examples/tpch/README.md:3]:
```suggestion
TPC-H is a benchmark suite for business-oriented ad-hoc queries that are used to simulate real questions and is usually used to benchmark the performance of database tools for answering them.
```

File: examples/tpch/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+`python generate_data_pq.py --SF 1 --folder SF1`
+
+Generate 1TB data and upload to S3 bucket: 
+
+`python generate_data_pq.py --SF 1000 --folder s3://bucket-name/`
+
+NOTES:
+
+- Script assumes `tpch-dbgen` is in the same directory. If you downloaded it in another location, make sure to update `tpch_dbgen_location` in the script with new location.
```
Comment (ritwika314) [inline examples/tpch/README.md:43]:
```suggestion
This script assumes `tpch-dbgen` is in the same directory. If you downloaded it at another location, make sure to update `tpch_dbgen_location` in the script with the new location.
```

File: examples/tpch/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+Run with 288 cores on S3 bucket data
+
+`mpiexec -n 288 python bodo_queries.py --folder s3://bucket-name/`
+
+## Spark
+
+### Installation
+
+Here, we show instructions for using PySpark with EMR cluster.
```
Comment (ritwika314) [inline examples/tpch/README.md:82]:
```suggestion
Here, we show the instructions for using PySpark with an EMR cluster. 
```

File: examples/tpch/README.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+## Spark
+
+### Installation
+
+Here, we show instructions for using PySpark with EMR cluster. 
+
+For other cluster configurations, please follow corresponding vendor's instructions.
+
+Follow steps outlined in "Launch an Amazon EMR cluster" section  of [AWS guide](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-gs-launch-sample-cluster.html)
```
Comment (ritwika314) [inline examples/tpch/README.md:86]:
```suggestion
Follow the steps outlined in the "Launch an Amazon EMR cluster" section of the [AWS guide](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-gs-launch-sample-cluster.html)
```

File: examples/tpch/README.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+In **Software configuration** step, make sure `Hadoop`, `Hive`, `JupyterEnterpriseGateway`, and `Spark` are selected.
```
Comment (ritwika314) [inline examples/tpch/README.md:88]:
```suggestion
In the **Software configuration** step, select `Hadoop`, `Hive`, `JupyterEnterpriseGateway`, and `Spark`.
```

File: examples/tpch/README.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+In **Cluster Nodes and Instances** step, choose same instance type for both master and workers. Don't create any task instances.
```
Comment (ritwika314) [inline examples/tpch/README.md:90]:
```suggestion
In the **Cluster Nodes and Instances** step, choose the same instance type for both master and workers. Don't create any task instances. 
```

File: examples/tpch/README.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+Follow steps outlined in "Launch an Amazon EMR cluster" section  of [AWS guide](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-gs-launch-sample-cluster.html)
+
+In **Software configuration** step, make sure `Hadoop`, `Hive`, `JupyterEnterpriseGateway`, and `Spark` are selected.
+
+In **Cluster Nodes and Instances** step, choose same instance type for both master and workers. Don't create any task instances. 
+
+### Running queries
+
+Attach [pyspark\_notebook.ipynb](./pyspark_notebook.ipynb) to your EMR cluster following the examples on [AWS documentation](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-managed-notebooks-create.html)
```
Comment (ritwika314) [inline examples/tpch/README.md:94]:
```suggestion
Attach [pyspark\_notebook.ipynb](./pyspark_notebook.ipynb) to your EMR cluster following the examples in the [AWS documentation](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-managed-notebooks-create.html)
```

**Pull request merged**